### ISSUE TITLE
K8s operator - imagePullSecrets are not propagated to proxy pods

### ISSUE THREAD

**Issue by thof** [tailscale/tailscale#10648]
### What is the issue?

In K8s operator you can specify the `imagePullSecrets` but it only applies to the operator pod. In result I see the following error for proxy pods:
```
429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit.
```

### Steps to reproduce

Just check the pod spec for proxy.

**Comment by irbekrm** [comment]
Thank you for opening the issue @thof .

> 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit.

Where are you pulling the images from, is it a custom registry of your own that needs a secret?

I think this issue might be because [we set the image pull policy on proxies to 'Always'](https://github.com/tailscale/tailscale/blob/v1.56.1/cmd/k8s-operator/deploy/manifests/proxy.yaml#L27). This was originally done because we set to the operator and proxy image tags to the static `unstable` tag- to ensure that the latest image version actually gets pulled. We now do publish stable versioned images for the operator and proxies. Our published Helm charts now use image versions, but we are still to publish versioned static manifests. Once we've done that, we can also change to image pull policy.

But I agree that we should allow propagating the image pull secret to proxies. Presumably folks always use the same registry for all images, so it makes sense to specifcy the secrets once for both operator and proxies.

**Comment by thof** [comment]
> Where are you pulling the images from, is it a custom registry of your own that needs a secret?

DockerHub https://docs.docker.com/docker-hub/download-rate-limit/ so if you don't use a paid subscription then it's very easy to reach the limit. That's why it's important to propagate the image pull secret to proxies.

**Comment by irbekrm** [comment]
Yes, I agree that we should have a way how to set them. We do want to create a mechanism for specifying additional configuration for proxies (i.e a custom resource like 'ProxyClass'.
But perhaps, for the image pull secrets, we can just add another env var to the operator deployment manifest and set it from the helm templates too.

**Comment by irbekrm** [comment]
Custom image pull secrets and other customization for proxy resources can now (v1.60) be applied via a ProxyClass custom resource https://tailscale.com/kb/1236/kubernetes-operator#cluster-resource-customization-using-proxyclass-custom-resource.

We are planning to eventually add more customization options to ProxyClass.
Feedback on whether it fits your workflow + what other fields you might need to configure (and why) is very welcome.

**Comment by thof** [comment]
Sorry, but I forgot to share the feedback. It works now, thanks for adding it!

**Comment by yaroze** [comment]
Hi guys!
Thanks for the workaround!
I can also confirm it's working fine now.

However, this might require us to change a bunch of Ingresses and create a CRS that's not part of our pipelines.
Any chance we can get it added to the helm chart values.yaml?

From my understanding, it requires modifying the operator itself, which I would be happy to do if I had the skills =/
